File: deno/send_message/src/functions/send_sms_twilio.ts
export default async function (
  variables: any,
  phoneNumber: string,
  message: string
) {
  const accountSID = variables["TWILIO_ACCOUNT_SID"];
  const authToken = variables["TWILIO_AUTH_TOKEN"];
  const sender = variables["TWILIO_SENDER"];

  if (!accountSID) {
    return {
      success: false,
      message: "TWILIO_ACCOUNT_SID is not set",
    };
  }
  if (!authToken) {
    return {
      success: false,
      message: "TWILIO_AUTH_TOKEN is not set",
    };
  }
  if (!sender) {
    return {
      success: false,
      message: "TWILIO_SENDER is not set",
    };
  }

  try {
    fetch(
      `https://api.twilio.com/2010-04-01/Accounts/${accountSID}/Messages.json`,
      {
        method: "POST",
        headers: {
          "Content-Type": "application/x-www-form-urlencoded",
          Authorization: `Basic ${btoa(`${accountSID}:${authToken}`)}`,
        },
        body: new URLSearchParams({
          To: phoneNumber,
          From: sender,
          Body: message,
        }),
      }
    );
  } catch (err) {
    return {
      success: false,
      message: err.message,
    };
  }

  return { success: true };
}

File: dotnet/SendMessage/src/SendSmsTwilio.cs
using System.Text;

namespace SendMessageFunction
{
public class SendSmsTwilio
{
    public static async Task<Dictionary<string, object>> SendSMS(Dictionary<string, string> variables, string? phoneNumber, string? message)
    {
        if (string.IsNullOrEmpty(phoneNumber))
        {
            return new Dictionary<string,object> {{"success", false} , {"message","No phone number provided"}};
        }

        if (string.IsNullOrEmpty(message))
        {
            return new Dictionary<string,object> {{"success", false} , {"message","No message provided"}};
        }

        string? accountSID = variables.TryGetValue("TWILIO_ACCOUNT_SID", out string? accountSIDValue) ? accountSIDValue : null;
        string? authToken = variables.TryGetValue("TWILIO_AUTH_TOKEN", out string? authTokenValue) ? authTokenValue : null;
        string? sender = variables.TryGetValue("TWILIO_SENDER", out string? senderValue) ? senderValue : null;

        if (string.IsNullOrEmpty(accountSID))
        {
            return new Dictionary<string,object> {{"success", false}, {"message","Missing Twilio account SID"}};
        }

        if (string.IsNullOrEmpty(authToken))
        {
            return new Dictionary<string,object> {{"success", false}, {"message","Missing Twilio auth token"}};
        }

        if (string.IsNullOrEmpty(sender))
        {
            return new Dictionary<string,object> {{"success", false}, {"message","Missing Twilio sender"}};
        }

        
        try
        {
            using (var httpClient = new HttpClient())
            {
            httpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(Encoding.ASCII.GetBytes($"{accountSID}:{authToken}")));

                var values = new Dictionary<string, string>
                {
                    { "From", sender },
                    { "To", phoneNumber },
                    { "Body", message }
                };

                var content = new FormUrlEncodedContent(values);

                var response = await httpClient.PostAsync($"https://api.twilio.com/2010-04-01/Accounts/{accountSID}/Messages.json", content);
                response.EnsureSuccessStatusCode();
                var responseString = await response.Content.ReadAsStringAsync();
                return new Dictionary<string, object> {{ "success" , true }, {"message", "Your message was sent" }};
                
            }
        }
        catch (Exception e)
        {
            Console.WriteLine(e);
            return new Dictionary<string,object>  {{ "success", false}, {"message", e.ToString() }};
        }

        
    }
}
}